src/docs/sphinx/dev_guide/state_manager.rst

```diff
@@ -0,0 +1,125 @@
+``initialize()`` method.  ``StateManager`` is implemented as a global singleton so that its
+contained data can be accessed from physics modules anywhere in the simulation.  The singleton
+is initialized with a non-owning reference to an ``axom::sidre::DataStore`` and an output directory
+to which data will be saved.
+
+.. note:: ``StateManager`` does not own its ``DataStore`` because Serac uses a single datastore to
+  store different kinds of data - that is, data unrelated to the state defined above.  In particular,
+  input file data is also stored in the per-simulation ``DataStore`` instance.
+
+Before any other kinds of state can be created, a mesh must be registered via ``SetMesh()``.
+In order for a restart to work properly, all state data must be owned by the underlying
+``StateManager``, so ownership of the mesh is transferred via a ``unique_ptr``.  In the case of multi-mesh
+simulations, a name should also be used to uniquely identify the mesh.
+
+Individual physics modules - that are of course based on these kinds of state - can now be constructed.
+In general, this process looks something like the following:
+
+  1. The physics module constructor accepts a mesh pointer and forwards it to the ``BasePhysics``
+     constructor. This parameter is required only in multi-mesh configurations and defaults to the
+     default mesh otherwise.  Although meshes also have string-valued tags associated with them,
+     a user of a physics module would find it more intuitive to pass the mesh pointer they wish to use.
+     Specifically, the meaning of a mesh parameter is much easier to discern than a string parameter.
+
+  #. The physics module creates its fields (e.g., temperature for a thermal conduction module) via
+     calls to ``StateManager::newState()``.  In addition to the ``serac::FiniteElementState`` constructor
+     arguments, this method also accepts a string-valued tag for the mesh with which the field is
+     associated.  The appropriate tag is a member of ``BasePhysics`` and initialized in the previous step.
+     FIXME: Should we provide a protected helper method in ``BasePhysics`` so derived modules don't need
+     to reference the member explicitly? Or perhaps ``StateManager::newState()`` et al should just take
+     a mesh pointer instead of a tag?
+
+  #. The ``FiniteElementState`` is then constructed and registered in the corresponding ``MFEMSidreDataCollection``.
+     The only tricky part about this process is the need for the underlying ``GridFunction`` to be allocated within
+     Sidre.  This required an additional option to the ``FiniteElementState`` constructor that leaves the vector
+     data uninitialized (aka ``nullptr``).  After we create the ``FiniteElementState`` we register its underlying
+     ``GridFunction`` in the ``MFEMSidreDataCollection`` and zero it out.
+
+  #. When the user wishes to save simulation state to disk, they can call ``outputState()`` on their physics module.
```

Comment:
We could delete the `outputState` method from the `BasePhysics` class. It makes sense that this would be associated with the state manager instead of the individual physics modules for the reason you described. I believe `outputState` predates the state manager anyway.

Reply:
`outputState` would still be useful for non-sidre output I think, e.g., via ParaView or something

Reply:
That's true. We'll have to think about if we still want to support non-sidre output.